ui/js/directives/treeherder/clonejobs.js

```diff
@@ -899,10 +899,15 @@ treeherder.directive('thCloneJobs', [
                         rsMap[resultSetId].rs_obj.platforms.forEach(function(platform) {
                             addAdditionalJobParameters(platform.groups);
                         });
-                        _.defer(
-                            generateJobElements,
-                            resultsetAggregateId,
-                            rsMap[resultSetId].rs_obj);
+                        $q.when(
+                            _.defer(
```

Comment:
I don't think this will do the when after the ``_.defer``.  I did a quick test, just to be sure and it was the other way around.  Since ``defer`` doesn't return a promise, it won't wait till it's done.  It'll just get the timer value from ``defer`` and pass that to the ``.when`` right away.  Here's the test I tried:

   var printIt = function(fleh) {
       console.log(fleh);
   };
   $q.when(
       _.defer(printIt, "first")
   ).then(function() {
       console.log("second");
   });